Chess/Assets/Scripts/ArrayChess/PieceMover.cs

```diff
@@ -54,83 +49,54 @@ public class PieceMover : MonoBehaviour
     private Vector3[] chessSquaresDistance;
 
     private float targetDistance = 1f;
-    private float loadtimer = 1f;
 
     void Start()
     {
         rend = GetComponent<Renderer>();
         pieceStartColor = rend.material.color;
-
+        chessSquares = GameObject.FindGameObjectsWithTag("square"); // Populates the chessboard array of all square positions 
     }
 
-
-
-
-
     void Update()
     {
-        if (!isInitialSquareRendererLoaded)
-        {
-            getSquareRenderers();
-        }
-
-        // add a if to load only once per turn.
+        //cacheSquareData(); TODO: Will be info on previous square, for now will be default colour
 
-
-        if (!isHover && isBoardArrayLoaded && isSquareRendLoaded && isColourUpdated) // Reset colour of piece and pieceSquare.
-        {
-
-            resetColors(false);
-            isColourUpdated = false;
-
-        }
-
-        if (!isBoardArrayLoaded) // Loads chessBoard.
+        if (Input.GetMouseButton(0)) // All permutations of mouse clicks go here
         {
-
-            findAllSquares();
-            isBoardArrayLoaded = true;
-        }
-
-        if (Input.GetMouseButton(0) && isSquareRendLoaded) // Clicking piece locks colour of movesquare until another piece is clicked.
-        {
-            //cacheSquareData();
             ray = Camera.main.ScreenPointToRay(Input.mousePosition);
             Physics.Raycast(ray, out hit);
-            rayHitStringParent = hit.transform.parent.name;
-            getSquareRenderers();
-            isCached = true;
-            
+            mouseClickPosition = hit.transform.parent.name;
 
+            updateNearestSquareCoords();
```

Comment:
I split up these two functions so that they run here one after the other. Its better to seperate them so it is now much clearer the steps that are happening in the Update function.

However, we have to be aware that sometimes there will be async issues, where the result of one function may run faster then the other one unintentially which may cause strange results.

Another thing to consider is if the running of one of these functions fail, its gonna mess up other things later on, but if it happens we'll come to that then!